docs/doxygen/developer/style.txt

```diff
@@ -0,0 +1,296 @@
+- The [GNU complexity](https://www.gnu.org/software/complexity/manual/complexity.html) of every function should be less than 9.
+- Deviations from [MISRA C:2012](https://en.wikipedia.org/wiki/MISRA_C) coding guidelines should be documented as comments in the code.
+
+@subsection guide_developer_styleguide_codingstyle_typeguidelines Type guidelines
+@brief Guidelines for variable types.
+- Only fixed-width integer types should be used. Exceptions for `bool` and types required by third-party APIs.
+- The default integer in the libraries should be 32 bits wide, i.e. `int32_t` or `uint32_t`.
+- Sizes and lengths should be represented with `size_t`, and Boolean variables with `bool`.
+
+@subsection guide_developer_styleguide_codingstyle_example Example File
+@brief An example file that follows the coding style rules.
+
+See @ref guide_developer_styleguide_naming for how to name the functions, variables, and macros.
+
+@code{c}
+/* Included headers are at the top of the file. The config file include is always first. */
+#include "config.h"
+
+/* Standard includes are immediately after the config file. They are sorted alphabetically.
+ * They use angle brackets <> around the file name. */
+
+/* Standard includes. */
+#include <assert.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <string.h>
+
+/* Library internal headers are included next. They use quotes "" around the file name. */
+
+/* Library internal include. */
+#include "private/library_internal.h"
+
+/* Error handling include (include only when needed). */
+#include "private/error.h"
+
+/* Library application-facing headers are included last. They use quotes "" around the file name. */
+
+/* Library include. */
+#include "library.h"
+
+/*-----------------------------------------------------------*/
+
+/* Defined constants follow the included headers. */
+
+/* When possible, parentheses () should be placed around constant values and a type
+ * should be specified. */
+#define LIBRARY_CONSTANT ( ( int32_t ) 10 )
+
+#define LIBRARY_FUNCTION_MACRO( argument ) \ /* Line continuators are right-aligned. */
+    {                                      \ /* Function-like macros are surrounded by curly braces {}. */
+        macro_body( ( argument ) );        \ /* Parentheses surround macro arguments for MISRA 20.7 */
+    }
+
+/*-----------------------------------------------------------*/
+
+/* Library typedefs follow the defined constants. */
+
+/* Forward declarations are used only when necessary. They are placed before all
+ * other typedefs. */
+
+typedef int32_t type_t;
+
+/**
+   @brief Structs are named along with the typedef. 
+*/
+typedef struct structType 
+{
+    int32_t member;
+
+    /* As usage of unions violates MISRA rule 19.2, we must document our usage
+     * and justification. Unions are used here to reduce the size of this struct. */
+    union /* Anonymous structs/unions are permitted only inside of other structs. */
+    {
+        int32_t a;
+        int32_t b;
```

Comment:
When would you ever have a union of two `int32_t`s...?

Reply:
Just copied from the wiki. I don't think we even allow unions, so I'm removing this.